packages/eas-build-job/src/step.ts

```diff
@@ -65,10 +70,11 @@ export const FunctionStepZ = CommonStepZ.extend({
    *      - value1
    *   arg4: ${{ steps.step1.outputs.test }}
    */
-  with: z.record(z.unknown()).optional(),
+  with: z.record(z.union([z.string(), z.number(), z.record(z.any())])).optional(),
```

Comment:
What did this change help with? I'm worried limiting this here may shoot us down the road. Example — YAML coalesces booleans to `z.boolean()`, so `a: true` will fail to be recognized as valid.

<img width="686" alt="Zrzut ekranu 2024-07-30 o 11 10 15" src="https://github.com/user-attachments/assets/e87afd20-c707-473d-8f5a-8b80cb691c48">

Reply:
The `z.record(z.union([z.string(), z.number(), z.record(z.any())]), z.boolean())` is the current type of accepted `inputs` https://github.com/expo/eas-build/blob/ad0487b2e5609aa879c6b488360d11442a45f542/packages/steps/src/BuildStepInput.ts#L19-L27

You are right I previously missed `boolean` and it was `z.record(z.union([z.string(), z.number(), z.record(z.any())]))` instead of `z.record(z.union([z.string(), z.number(), z.record(z.any())]), z.boolean())` 👍